protoc.mk

```diff
@@ -0,0 +1,28 @@
+# This Makefile makes it easy for the language bindings
+# to download and install the protoc binary.
+
+PROTOC_OS := $(shell uname -s)
+ifeq (Darwin,$(PROTOC_OS))
+PROTOC_OS := osx
+endif
+
+PROTOC_ARCH := $(shell uname -m)
+ifeq (i386,$(PROTOC_ARCH))
+PROTOC_ARCH := x86_32
+endif
+
+PROTOC := ./protoc
```

Comment:
I only made `proto.mk` external from the Go process in the case that you might want to re-use it. If that's not the case I'll move it into `lib/go/Makefile` directly. Thank you for the feedback.